numba/tests/test_ir_inlining.py

```diff
@@ -526,6 +529,48 @@ def impl():
 
         self.check(impl, inline_expect={'foo': True})
 
+    # adapted from #5064
+    def test_inline_always_operators(self):
+
+        DummyType = type('DummyType', (types.Opaque,), {})
+
+        dummy_type = DummyType("my_dummy")
+        register_model(DummyType)(OpaqueModel)
+
+        class Dummy(object):
+            def __eq__(self, other):
+                return True
+
+            def __getitem__(self, idx):
+                return None
+
+        @typeof_impl.register(Dummy)
+        def typeof_dummy(val, c):
+            return dummy_type
+
+        @unbox(DummyType)
+        def unbox_dummy(typ, obj, c):
+            return NativeValue(c.context.get_dummy_value())
+
+        @overload(operator.eq, inline='always')
+        def overload_dummy_eq(a, b):
+            if a == dummy_type:
+                return lambda a, b: True
+
+        @overload(operator.getitem, inline='always')
+        def overload_dummy_getitem(a, idx):
+            if a == dummy_type:
+                return lambda a, idx: None
+
+        def impl_eq(x):
+            return x == 1
+
+        def impl_getitem(x):
+            return x[1]
+
+        self.check(impl_eq, Dummy(), inline_expect={'eq': True})
+        self.check(impl_getitem, Dummy(), inline_expect={'getitem': True})
```

Comment:
Given the scope of what `is_operator` is covering, some more tests would be good. Would suggest covering at least one each of: `['unary', 'binop', 'inplace_binop', 'getitem', 'static_getitem']`.